### PR TITLE
terrianx/58-mission

### PR DESCRIPTION
**Pull request by terrianx**
MVP draft for missions section. not mobile responsive

overflow-x-hidden necessary to hide horizontal scroll because of solution needed to address small gap < 1px wide between triangle and text (see picture attached). resize chrome browser window to see gap appear and reappear

solution is a negative 1px margin for middle text block section, which necessitates altering triangle polygons to have 99% and 1% to cover up and show clean sharp corners like in the mockup
Screenshot 2023-08-05 at 11 08 22 PM

quick view of component
Screenshot 2023-08-06 at 12 19 58 AM

### CODE REVIEW

**Review by thedomcab — COMMENTED** [review]
With the triangle, you can use svg instead background clip which is more cleaner html and css. look at the way I did in partners.js.

**Comment by terrianx** [comment]
> With the triangle, you can use svg instead background clip which is more cleaner html and css. look at the way I did in partners.js.

I saw what you did and it is cleaner but the text overflows when resizing the window. The polygon solution with clip path guarantees that the text cannot overflow and is more responsive for different window sizes. However, if you dont think this is an issue, I can rewrite the component. 

Another solution more similar to the SVG one is to make the entire background a trapezoid instead of splitting it up into triangles and a text box. I have explored this solution already and it is quite clean and requires only one line of css, but runs into the same issues due to text overflowing out of the darker blue area.

**Comment by thedomcab** [comment]
May you give me a demo on what you mean? I can't understand what you mean.

**Comment by terrianx** [comment]
> May you give me a demo on what you mean? I can't understand what you mean.

Actually I redesigned it and solved the issue with the latest commit. text is guaranteed to not flow over now. i like this solution since we dont need to add an svg and everything can just be done with css. it is also a lot more responsive now to resizing

notice how the text does not flow over.
Screenshot 2023-08-06 at 11 51 56 PM

this text flows over
Screenshot 2023-08-06 at 11 52 09 PM

**Comment by thedomcab** [comment]
Why are you so against svgs and you can use borders instead of clip bath which has the same effect.  Which we can use the tailwindcss only without any custom css not available in tailwindcss

**Comment by thedomcab** [comment]
Having an empty divs are not best practices to implement web design

**Comment by terrianx** [comment]
> Why are you so against svgs and you can use borders instead of clip bath which has the same effect. Which we can use the tailwindcss only without any custom css not available in tailwindcss

i think svgs can be useful but for this case text can flow over it. borders would need to have their width specified which may make it more complex to add responsiveness. i understand that it may not be ideal to use css outside of tailwind. i do also agree that empty divs are not good, but in this case they are filled with the content and wrap the text in easy to move around layers such as text, trapezoids, and boxes.

given the problems ive noticed with using svgs and borders, i believe using clip path is the best solution for functionality, maintainability, and scalability. the solution works well and is already implemented, but moving forward i will keep using exclusively tailwind in mind.

if you think otherwise, id be willing to rewrite the component using the svg solution after discussing more or with another reviewer. from what i understand about MVPs, we are not chasing perfection and a solution good enough should work.

i understand that i may not be as experienced as you in frontend development, so i am very open to constructive and justified feedback.

**Comment by thedomcab** [comment]
I got it to work with borders.  It not too difficult.  I have over 5 years of experience working with responsive web design.  You can make borders overflow the the container which make it responsive.

**Comment by terrianx** [comment]
i will use borders, thanks for the suggestion

**Comment by thedomcab** [comment]
Don't forget about pseudo classes you can use without adding additional divs.  That is the reason w3c added this things to css without using unnecessary divs.  Css has so much capabilities which they are always changing to less the load on adding unnecessary html and use more css capabilities

**Comment by terrianx** [comment]
> I got it to work with borders. It not too difficult. I have over 5 years of experience working with responsive web design. You can make borders overflow the the container which make it responsive.

ok i have been trying the border solution but you need to set exact pixels for the properties which is hard coding values. look at picture 1. i also found non tailwind css in the globals.css. wouldnt it also be ok to use nontailwind for this component? 

i want to emphasize that the solution with clip path works very well and it looks very much like the figma. can you help me with this component or should i reach out to someone else? i cant get the border solution to work.

picture 1
Screenshot 2023-08-09 at 12 03 33 AM


Screenshot 2023-08-09 at 12 01 39 AM
Screenshot 2023-08-09 at 12 06 47 AM

File: frontend/src/css/mission.module.css [hunk 1]
```diff
@@ -0,0 +1,51 @@
+@config "../config/tailwind.config.js";
+
+.container {
+  @apply pt-20 h-full w-screen relative left-1/2 right-1/2 -mx-[50vw];
+}
+
+.top-box {
+  @apply bg-Primary-500;
+}
```
Comment (thedomcab) [inline frontend/src/css/mission.module.css:9]:
```suggestion
.top-box {
  @apply bg-Primary-900 relative min-h-[25rem] flex justify-center items-center;
}

.top-box p {
  @apply max-w-[32.5rem];
}

.top-box::before,
.top-box::after {
  @apply content-[""] absolute top-0  border-solid border-t-transparent border-t-[25rem];
}

.top-box:before {
  @apply border-l-Primary-500 border-l-[1.875rem] lg:border-l-[25rem] xl:border-l-[31.25rem] left-0;
}

.top-box::after {
  @apply border-r-Primary-500 border-r-[1.875rem] lg:border-r-[25rem] xl:border-r-[31.25rem] right-0;
}
```

File: frontend/src/css/mission.module.css [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+.bottom-box {
+  @apply bg-Primary-300;
+}
+
```
Comment (thedomcab) [inline frontend/src/css/mission.module.css:14]:
```suggestion

.bottom-box {
  @apply bg-Primary-700 relative min-h-[20rem] flex justify-center items-center;
}

.bottom-box p {
  @apply max-w-[33.75rem];
}

.bottom-box::before,
.bottom-box::after {
  @apply content-[""] absolute top-0 border-solid border-t-Primary-300 border-t-[20rem];
}

.bottom-box::before {
  @apply border-r-transparent border-r-[1.875rem] lg:border-r-[25rem] xl:border-r-[31.25rem] left-0;
}

.bottom-box::after {
  @apply border-l-transparent border-l-[1.875rem] lg:border-l-[25rem] xl:border-l-[31.25rem] right-0;
}
```

File: frontend/src/css/mission.module.css [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  @apply text-6xl text-Shades-0 whitespace-nowrap;
+}
+
+.text p {
+  @apply font-bold text-xl pt-5 max-w-xl;
+}
+
+.text span {
+  @apply text-2xl;
+}
```
Comment (thedomcab) [inline frontend/src/css/mission.module.css:39]:
```suggestion
.text p {
  @apply font-bold text-xl pt-5 mb-0;
}

.text span {
  @apply text-2xl leading-tighter;
}
```

**Review by thedomcab — APPROVED** [review]

**Pull request merged**